app/models/request.rb

```diff
@@ -156,7 +156,7 @@ def date_for_state(state)
                   'INNER JOIN uuids ON uuids.resource_id=requests.submission_id AND uuids.resource_type="Submission"'
                 ]
             )
-            .group('uuids.external_id')
+            .group('pool_id')
```

Comment:
can you group by selection alias here?

Reply:
As far as I know it's valid SQL syntax and it does give the same output.  Plus it makes it clearer what you're actually grouping by.

Reply:
I think aliasing happens after grouping. I just made up the below and it gives an error with group by pool_id
`Error Code: 1054. Unknown column 'pool_id' in 'group statement'`
but it works with uuids.external_id . 


```
select "uuids.external_id AS pool_id,
              GROUP_CONCAT(DISTINCT pw_location.description ORDER BY pw.map_id ASC SEPARATOR \",\") AS pool_into,
              SUM(requests.state = 'passed') > 0 AS pool_complete,
              MIN(requests.id) AS id,
              MIN(requests.sti_type) AS sti_type,
              MIN(requests.target_asset_id) AS target_asset_id,
              MIN(requests.submission_id) AS submission_id,
              MIN(requests.request_type_id) AS request_type_id"
from requests  
JOIN receptacles AS pw ON requests.asset_id=pw.id
JOIN well_links ON well_links.source_well_id=requests.asset_id
join maps AS pw_location ON pw.map_id=pw_location.id
join  uuids ON uuids.resource_id=requests.submission_id AND uuids.resource_type="Submission"
group by pool_id
limit 10;
```

Reply:
I just ran the following query (which Rails generated from the code shown) and it works fine:

```
SELECT uuids.external_id AS pool_id,
              GROUP_CONCAT(DISTINCT pw_location.description ORDER BY pw.map_id ASC SEPARATOR ",") AS pool_into,
              SUM(requests.state = 'passed') > 0 AS pool_complete,
              MIN(requests.id) AS id,
              MIN(requests.sti_type) AS sti_type,
              MIN(requests.target_asset_id) AS target_asset_id,
              MIN(requests.submission_id) AS submission_id,
              MIN(requests.request_type_id) AS request_type_id FROM `requests` INNER JOIN well_links ON well_links.source_well_id=requests.asset_id INNER JOIN receptacles AS pw ON well_links.target_well_id=pw.id AND well_links.type="stock" INNER JOIN maps AS pw_location ON pw.map_id=pw_location.id INNER JOIN uuids ON uuids.resource_id=requests.submission_id AND uuids.resource_type="Submission" WHERE `pw`.`labware_id` = 13821 AND `requests`.`submission_id` = 1463 AND `requests`.`state` != 'cancelled' GROUP BY pool_id
```